wheel/python/clvm_rs/tree_hash.py

```diff
@@ -0,0 +1,107 @@
+"""
+This is an implementation of `sha256_treehash`, used to calculate
+puzzle hashes in clvm.
+
+This implementation goes to great pains to be non-recursive so we don't
+have to worry about blowing out the python stack.
+"""
+
+from hashlib import sha256
+from typing import Callable, List, Tuple, cast
+
+from .clvm_storage import CLVMStorage
+
+
+OP_STACK_F = Callable[[List[CLVMStorage], List[bytes], List["OP_STACK_F"]], None]
+
+
+class Treehasher:
+    """
+    `Treehasher` performs the standard sha256tree hashing in a non-recursive
+    way so that extremely large objects don't blow out the python stack.
+
+    We also force a `_cached_sha256_treehash` into the hashed sub-objects
+    whenever possible so that taking the hash of the same sub-tree is
+    more efficient in future.
+    """
+
+    atom_prefix: bytes
+    pair_prefix: bytes
+    cache_hits: int
```

Comment:
I don't see any use of `cache_hits` (but maybe it's in a collapsed diff)

Reply:
It was used during development. It might be good for tests or benchmarks to ensure that fast code paths are taken when expected. I can remove it, but I'd expect it not to have much performance impact.